File: beginner_tutorials/scripts/listener.py
#! /usr/bin/env python3

import rospy
from std_msgs.msg import String
from beginner_tutorials.msg import Num


def std_cb(data):
    tmp_str = "%s : i received %s" %(rospy.get_caller_id(), data.data)
    rospy.loginfo(tmp_str)

def num_cb(data):
    tmp_str = "%s : i received %s" %(rospy.get_caller_id(), data.num)
    rospy.loginfo(tmp_str)




def listener():
    rospy.init_node("beg_listener", anonymous=True)

    rospy.Subscriber("chatter", String, std_cb)
    rospy.Subscriber("num", Num, num_cb)

    rospy.spin()








if __name__ == "__main__":
    listener()

File: beginner_tutorials/src/listener.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>
#include <beginner_tutorials/Num.h>

void std_cb(const std_msgs::String::ConstPtr &msg) {
    ROS_INFO("i received: %s", msg->data.c_str());
}

void num_cb(const beginner_tutorials::Num::ConstPtr &msg) {
    ROS_INFO("i received: %ld", msg->num);
}

int main(int argc, char **argv) {
    ros::init(argc, argv, "example_listener");
    ros::NodeHandle nh;

    ros::Subscriber std_sub = nh.subscribe("chatter", 1000, &std_cb);
    ros::Subscriber num_sub = nh.subscribe("num", 1000, &num_cb);


    ros::spin();


    return 0;
}